docs/extend/components.md

```diff
@@ -109,8 +111,386 @@ const AnotherComponent = Ractive.extend({
+
+```html
+<child person={{client}}/>
+```
+
+We map the parent *client* object for the child by specifying an attribute *person* and assigning the *client* object to it.
+
+With the **data mapping** in place the child can now use the reference *person.name* in a mustache as ```{{person.name}}``` and the parent value of *client.name* will be rendered.
+
+Now we can see that the *person* in child is mapped from the parent.
+
+If we are using **unmanaged** components we can specify the **data maping** on the **anchor** tag:
+
+```html
+<#child person={{client}}/>
+```
+
+If you don't mind mapping the data on parent and child to the same name you can use a short-hand way for the data mapping:
+
+```html
+<child bind-client/>
+```
+Here the data is mapped to the child as *client*, so the child template will be ```{{client.name}}``` to render the client name.
+
+Using the **bind-"name"** notation on the element Ractive will map an object in the parent with the given *name* to the child under the same *name*.
+
+```html
+<comp person={{person}}/>
+```
+
+#### findComponent
+
+Ractive provides the method  [findComponent()](../api/instance-methods.md#ractivefindcomponent) to lookup a child instance registered under a given name. Once we have a reference to the child we can communicate with the child through it's APIs.
+
+```js
+let child = Ractive.extend({
+
+    isPersonValid: function( ) {
+        // Perform validation on person object
+        var person = child.get('person');
+        if ( person.name == '' ) {
+            return false;
+        }
+        return true;
+    }
+});
+
+let parent= new Ractive({
+
+    save: function() {
+
+        // lookup the child component
+        var child = this.findComponent('child');
+
+        // validate person using child API
+        if (child.isPersonValid()) {
+
+            var person = child.get('person');
+            // save person to database
+        }
+    }
+
+    components: {child: child}
+});
+```
+
+In the example above the child has a method *isPersonValid* which forms part of the child API.
+
+The parent can call this API by looking up the child instance through [findComponent()](../api/instance-methods.md#ractivefindcomponent).
+
+#### Fire events
+We mentioned earlier that the child should not directly reference the parent but should  [fire()](../api/instance-methods.md#ractivefire) events that the parent can listen to.
+
+Say the child has a save button the user can click once done editing a *person* object. When the *save* button is clicked we want to inform listeners that the *person* object has changed, so we **fire** the event **personChanged** passing the parent object as an argument.
+
+The parent component can register as a listener to receive the **personChanged** event.
+
+Let's look at how we can do that.
+
+Here is the child template:
+
+```html
+{{#persons}}
+<input value="name" type="text"/>
```

Comment:
Thanks! Because of two-way binding the person object in parent would be updated anyway unless twoway is set to false. Practically though, switching off twoway is painful. But hopefully the idea of firing events is still conveyed. Interestingly, doing homework on other libraries sporting one way data flow they have caveats that binding on 'objects' updates the parent anyway.